File: backend/src/main/java/com/ums/ums_backend/entity/Faculty.java
package com.ums.ums_backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
public class Faculty {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(nullable = false, length = 10)
    private String code;

    @Column(length = 250)
    private String description;

    @OneToMany(mappedBy = "faculty")
    private List<Department> departments;

}

File: frontend/src/app/features/faculty/models/faculty.ts
import {Department} from '../../department/models/department';

export interface Faculty {

  id: number,
  name: string,
  code: string,
  description: string,
  departments: Department[]

}

export interface FacultyCreateRequest {

  name: string,
  code: string,
  description: string,

}
